extra_scripts/codemirror/addon/hyperlink/hyperlink.js

```diff
@@ -60,13 +74,83 @@
 
       return null
     }
+    specialLinkHandler(e, rawHref, linkHash) {
+      const isStartWithHash = rawHref[0] === '#'
+
+      const extractIdRegex = /file:\/\/.*main.?\w*.html#/ // file://path/to/main(.development.)html
+      const regexNoteInternalLink = new RegExp(`${extractIdRegex.source}(.+)`)
+      if (isStartWithHash || regexNoteInternalLink.test(rawHref)) {
+        const posOfHash = linkHash.indexOf('#')
+        if (posOfHash > -1) {
+          const extractedId = linkHash.slice(posOfHash + 1)
+          const targetId = mdurl.encode(extractedId)
+          const targetElement = document.getElementById(targetId) // this.getWindow().document.getElementById(targetId)
+
+          if (targetElement != null) {
+            this.scrollTo(0, targetElement.offsetTop)
+          }
+          return
+        }
+      }
+
+      // this will match the new uuid v4 hash and the old hash
+      // e.g.
+      // :note:1c211eb7dcb463de6490 and
+      // :note:7dd23275-f2b4-49cb-9e93-3454daf1af9c
+      const regexIsNoteLink = /^:note:([a-zA-Z0-9-]{20,36})$/
+      if (regexIsNoteLink.test(linkHash)) {
+        eventEmitter.emit('list:jump', linkHash.replace(':note:', ''))
+        return
+      }
+
+      const regexIsLine = /^:line:[0-9]/
+      if (regexIsLine.test(linkHash)) {
+        const numberPattern = /\d+/g
+
+        const lineNumber = parseInt(linkHash.match(numberPattern)[0])
+        eventEmitter.emit('line:jump', lineNumber)
+        return
+      }
+
+      // this will match the old link format storage.key-note.key
+      // e.g.
+      // 877f99c3268608328037-1c211eb7dcb463de6490
+      const regexIsLegacyNoteLink = /^(.{20})-(.{20})$/
+      if (regexIsLegacyNoteLink.test(linkHash)) {
+        eventEmitter.emit('list:jump', linkHash.split('-')[1])
+        return
+      }
+
+      const regexIsTagLink = /^:tag:#([\w]+)$/
```

Comment:
Can you remove the `#` since we decided to remove `#` from tag link in the previous PR?

Reply:
I've removed it but there is a new issue introduced by this change - see comment below.